ansible/playbooks/roles/kubernetes_master/tasks/master-untaint.yml

```
@@ -1,32 +1,37 @@
 ---
 - name: Get masters with NoSchedule
-  vars:
-    selector: >-
-      'node-role.kubernetes.io/master'
-    jsonpath: >-
-      '{range .items[*]}{.metadata.name}{":"}{.spec.taints[?(@.effect == "NoSchedule")].effect}{"\n"}{end}'
-  shell: |
-    kubectl get nodes \
-      --selector={{ selector }} \
-      --output=jsonpath={{ jsonpath }} \
-    | grep :NoSchedule
+  command: >-
+    kubectl get nodes --output go-template='
+    {{ '{{' }}range $item := .items{{ '}}' }}
+    {{ '{{' }}range $taint := $item.spec.taints{{ '}}' }}
+    {{ '{{' }}if and ( or ( eq $taint.key "node-role.kubernetes.io/master")
+    (eq $taint.key "node-role.kubernetes.io/control-plane") )
+    (eq $taint.effect "NoSchedule"){{ '}}' }}
+    {{ '{{' }}printf "%s\n" $item.metadata.name{{ '}}' }}
+    {{ '{{' }}end{{ '}}' }}
+    {{ '{{' }}end{{ '}}' }}
+    {{ '{{' }}end{{ '}}' }}'
   register: tainted_masters_list
-  failed_when: tainted_masters_list.rc == 2
   changed_when: false
 
-- name: Decide if we want/need to untaint masters
+- name: Untaint all tainted masters
   vars:
-    nodeless: >-
-      {{ groups.kubernetes_node | default([]) | length == 0 }}
-  set_fact:
-    untaint_master: >-
-      {{ (tainted_masters_list.stdout_lines | length > 0) and (nodeless or specification.allow_pods_on_master) }}
+    # Filter empty lines out
+    tainted_control_plane_nodes: >-
+      {{ tainted_masters_list.stdout_lines | map('trim') | list | select() | list }}
```

```suggestion
      {{ tainted_masters_list.stdout_lines | map('trim') | select }}
```
Casting to `list` is not required in current Ansible (it was in the past)